var/spack/repos/builtin/packages/hdf5-vol-async/package.py

```diff
@@ -6,6 +6,20 @@
 from spack.package import *
 
 
+def require_mpi_thread_multiple(env=None):
+    if env is not None:
+        # Configure this for MPICH and the many MPICH forks
+        env.set("MPICH_MAX_THREAD_SAFETY", "multiple")
+    else:
+        # Don't allow building with MPIs that don't support
+        # MPI_THREAD_MULTIPLE
+        msg = "HDF5 VOL Async requires MPI with MPI_THREAD_MULTIPLE support"
+        conflicts("openmpi ~thread_multiple", msg=msg)
```

Comment:
I was playing around with the concretizer.  It seems you can actually get it to work with `spack spec -I ecp-data-vis-sdk +hdf5 ^openmpi~thread_multiple` because `thread_multiple` variant is not enabled for all versions.  I believe this still does exactly what we need it to.

Reply:
```
 PKG-DIRECTIVES: 2 issues found
1. hdf5-vol-async: wrong variant in "conflicts" directive
    the variant 'thread_multiple' does not exist
    in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/hdf5-vol-async/package.py
2. hdf5-vol-async: wrong variant in "conflicts" directive
    the variant 'threads' does not exist
    in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/hdf5-vol-async/package.py
Error: Process completed with exit code 1.
```